+nosnoc/+discrete_time_problem/Gcs.m

```diff
@@ -0,0 +1,870 @@
+                obj.f = obj.f + obj.w.T_final();
+            end
+
+            % 1d vars: Variables that are defined per control stage
+            obj.w.u(1:opts.N_stages) = {{'u', dims.n_u}, model.lbu, model.ubu, model.u0};
+            obj.p.p_time_var(1:opts.N_stages) = {{'p_time_var', dims.n_p_time_var}, model.p_time_var_val};
+            if opts.use_speed_of_time_variables && opts.local_speed_of_time_variable
+                obj.w.sot(1:opts.N_stages) = {{'sot', 1}, opts.s_sot_min, opts.s_sot_max, opts.s_sot0};
+            end
+
+            % TODO(@anton) This _severely_ hurts performance over the vectorized assignment by doing N_stages vertcats of
+            %              casadi symbolics vs just a vectorized assignment which does one. As such there needs to be backend
+            %              work done for vdx to cache vertcats of SX somehow. Current theory is one can simply keep a queue of
+            %              symbolics to be added in a cell array until a read is done, at which point we call a single vertcat
+            %              on the whole queue which is _significantly_ faster.
+            % 2d vars: Variables that are defined for each finite element.
+            for ii=1:opts.N_stages
+                % other derived values
+                h0 = opts.h_k(ii);
+                if obj.opts.use_fesd
+                    ubh = (1 + opts.gamma_h) * h0;
+                    lbh = (1 - opts.gamma_h) * h0;
+                    if opts.time_rescaling && ~opts.use_speed_of_time_variables
+                        % if only time_rescaling is true, speed of time and step size all lumped together, e.g., \hat{h}_{k,i} = s_n * h_{k,i}, hence the bounds need to be extended.
+                        ubh = (1+opts.gamma_h)*h0*opts.s_sot_max;
+                        lbh = (1-opts.gamma_h)*h0/opts.s_sot_min;
+                    end
+                    obj.w.h(ii,1:opts.N_finite_elements(ii)) = {{'h', 1}, lbh, ubh, h0};
+                end
+                if obj.opts.step_equilibration == StepEquilibrationMode.linear_complementarity
+                    obj.w.B_max(ii,2:opts.N_finite_elements(ii)) = {{'B_max', dims.n_lambda},-inf,inf};
+                    obj.w.pi_c(ii,2:opts.N_finite_elements(ii)) = {{'pi_c', dims.n_c},-inf,inf};
+                    obj.w.pi_lambda(ii,2:opts.N_finite_elements(ii)) = {{'pi_lambda', dims.n_lambda},-inf,inf};
+                    obj.w.lambda_c(ii,2:opts.N_finite_elements(ii)) = {{'lambda_c', dims.n_c},0,inf};
+                    obj.w.lambda_lambda(ii,2:opts.N_finite_elements(ii)) = {{'lambda_lambda', dims.n_lambda},0,inf};
+                    obj.w.eta(ii,2:opts.N_finite_elements(ii)) = {{'eta', dims.n_lambda},0,inf};
+                    obj.w.nu(ii,2:opts.N_finite_elements(ii)) = {{'nu', 1},0,inf};
+                end
+                if ~opts.right_boundary_point_explicit
+                    obj.w.not_c_primals(ii,1:opts.N_finite_elements(ii)) = {{'not_c_primal', 2}, 0 inf, 0.5};
```

Comment:
having for the default hard coded numbers that one cannot change easily is maybe not nice

Reply:
Technically the nice part of the vdx backend is changing the numbers isn't that hard after the fact (if a little detailed)
```
ocp_solver.discrete_time_problem.w.not_c_primals(1,2).init  = 5;
```
for example. 

However in this case there do need to be changes here made to allow for `n_c  > 1`.

Reply:
But we dont expect everyone interested in only in the fronted to know the internally used vdx syntax. 
As for any bounds, initial values and co, I prefer having variables stored somewhere (This is also consistent with the currently new design existing in main)

Reply:
This particular issue will be gone due to above rbp discussion. However I generally agree that backend vdx syntax should be hidden. As such I have a TODO to make a robust `.set` method for `nosnoc.Integrator` and `nosnoc.ocp.Solver` which should hide the vdx syntax.

Reply:
Probably should be an issue. Added #110